travis/shared_configs/doctr-upload.yml

```
@@ -0,0 +1,39 @@
+jobs:
+  include:
+    - stage: deploy
+      name: "Docs Upload"
+      install:
+
+        - echo "Req File':' ${DOCS_REQUIREMENTS:=docs-requirements.txt}"
+        - echo "Folder':' ${DOCS_FOLDER:=docs}"
+        - |
+          # Install docs requirements
+          if [[ -f "${DOCS_REQUIREMENTS}" ]]; then
+            pip install -Ur ${DOCS_REQUIREMENTS}
```

While there's nothing technically wrong with this, what do you think about verbose `--options` in these scripts? Saving a few bytes over readability doesn't seem like the right priority:

```suggestion
            pip install --upgrade --requirements "${DOCS_REQUIREMENTS}"
```

Failing that, always quote paths just in case there are spaces in them:
```suggestion
            pip install -Ur "${DOCS_REQUIREMENTS}"
```

---

I'm a little confused when `--upgrade` is necessary. I guess my concerns are:
- If not used, will old versions of packages not be updated?
- If used, will specified packages that aren't already downloaded not be installed?

I assume I should always use it when installing these requirements files?

---

There might not be any issue removing it. The intent, at least, seems clear-ish: "please ensure that everything is up-to-date when you install these packages".

This reminds me: do we have a `pip install --upgrade pip` anywhere? I've run into issues in the past where old pip versions can be problematic, especially when relying on the system (or CI image) python and not our conda Python versions.

---

Fixed in recent commits.